Submitted by: Matt Burgess (matthew at linuxfromscratch.org)
Date: 2008-02-03
Initial Package Version: 6.10
Upstream Status: From Upstream
Origin: http://lists.gnu.org/archive/html/bug-automake/2008-01/msg00036.html

diff -Naur automake-1.10.1.orig/tests/instsh2.test automake-1.10.1/tests/instsh2.test
--- automake-1.10.1.orig/tests/instsh2.test	2008-01-21 22:11:41.000000000 +0000
+++ automake-1.10.1/tests/instsh2.test	2008-02-03 21:26:17.000000000 +0000
@@ -1,5 +1,5 @@
 #! /bin/sh
-# Copyright (C) 2002, 2004, 2006  Free Software Foundation, Inc.
+# Copyright (C) 2002, 2004, 2006, 2008  Free Software Foundation, Inc.
 #
 # This file is part of GNU Automake.
 #
@@ -81,20 +81,6 @@
 ./install-sh -T x d3 && exit 1
 ./install-sh -T x d4// && exit 1
 
-# Do not change the timestamps when using -C.
-echo foo >file
-./install-sh -C file d1
-TZ=UTC0 touch -t $old_timestamp d1/file
-./install-sh -C file d1
-is_newest file d1/file
-echo foo1 >file
-./install-sh -C file d1
-diff file d1/file
-# Rights must be updated.
-./install-sh -C -m 444 file d1
-test -r d1/file
-test ! -w d1/file
-
 # Ensure that install-sh works with names that include spaces
 touch 'a  b'
 mkdir 'x  y'
diff -Naur automake-1.10.1.orig/tests/instsh3.test automake-1.10.1/tests/instsh3.test
--- automake-1.10.1.orig/tests/instsh3.test	1970-01-01 00:00:00.000000000 +0000
+++ automake-1.10.1/tests/instsh3.test	2008-02-03 21:26:17.000000000 +0000
@@ -0,0 +1,37 @@
+#! /bin/sh
+# Copyright (C) 2006, 2008  Free Software Foundation, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# More install-sh checks: check -C.
+
+required=non-root
+. ./defs || exit 1
+set -e
+
+./install-sh -d d1
+
+# Do not change the timestamps when using -C.
+echo foo >file
+./install-sh -C file d1
+TZ=UTC0 touch -t $old_timestamp d1/file
+./install-sh -C file d1
+is_newest file d1/file
+echo foo1 >file
+./install-sh -C file d1
+diff file d1/file
+# Rights must be updated.
+./install-sh -C -m 444 file d1
+test -r d1/file
+test ! -w d1/file
